util/vendor.py

```diff
@@ -336,115 +440,97 @@ def main(argv):
     else:
         log.basicConfig(format="%(levelname)s: %(message)s")
 
-    desc_file_path = Path(args.desc_file.name).resolve()
-    vendor_file_base_dir = desc_file_path.parent
-
-    # Precondition: Ensure description file matches our naming rules
-    if not str(desc_file_path).endswith('.vendor.hjson'):
-        log.fatal("Description file names must have a .vendor.hjson suffix.")
-        raise SystemExit(1)
-
-    # Precondition: Check for a clean working directory when commit is requested
+    # Load input files (desc file; lock file) and check syntax etc.
+    try:
+        # Load description file
+        desc = Desc(args.desc_file)
+        lock_file_path = desc.lock_file_path()
+
+        # Try to load lock file (which might not exist)
+        try:
+            with open(lock_file_path, 'r') as lock_file:
+                lock = LockDesc(lock_file)
+        except FileNotFoundError:
+            lock = None
+    except (JsonError, ValueError) as err:
+        log.fatal(str(err))
+        raise SystemExit(1) from None
```

Comment:
Oh, none probably. I think I added it on automatic: as you probably know, it matters with other exception types (to avoid chaining the backtraces). I'll get rid of it.